mk/config.mk

```diff
@@ -909,15 +909,28 @@ CFG_REGULATOR_FIXED ?= n
 $(eval $(call cfg-enable-all-depends,CFG_REGULATOR_FIXED, \
 	 CFG_DRIVERS_REGULATOR CFG_DT))
 
-# The purpose of this flag is to show a print when booting up the device that
-# indicates whether the board runs a standard developer configuration or not.
+# When enabled, CFG_INSECURE permits insecure configuration of OP-TEE core
+# and shows a print (info level) when booting up the device that
+# indicates that the board runs a standard developer configuration.
+#
 # A developer configuration doesn't necessarily has to be secure. The intention
 # is that the one making products based on OP-TEE should override this flag in
 # plat-xxx/conf.mk for the platform they're basing their products on after
 # they've finalized implementing stubbed functionality (see OP-TEE
 # documentation/Porting guidelines) as well as vendor specific security
 # configuration.
-CFG_WARN_INSECURE ?= y
+CFG_INSECURE ?= y
+
+# CFG_WARN_INSECURE serves the same purpose as CFG_INSECURE but is deprecated
+# and should not be used. CFG_WARN_INSECURE will be removed in a future OP-TEE
+# release.
+ifneq (,$(CFG_WARN_INSECURE))
+$(info CFG_WARN_INSECURE is deprecated, please use CFG_INSECURE instead)
+else
+ifneq ($(CFG_WARN_INSECURE),$(CFG_INSECURE))
+$(error Deprecated CFG_WARN_INSECURE must match CFG_INSECURE)
```

Comment:
This doesn't work due to "" != "n" but having the same meaning (disabled).
I don't think this is a proper thing to do. How about simply setting the default from the previous variable? 
```
CFG_INSECURE ?= $(CFG_WARN_INSECURE)
CFG_INSECURE ?= y
```

Reply:
I pushed a buggy commit here.
Please re-review with the latest fixup commit applied.

Reply:
I still believe that anyone setting `CFG_WARN_INSECURE` to disabled ("" or "n") would want `CFG_INSECURE` disabled, too, because they are confident that they platform is secure. Therefore `$(CFG_WARN_INSECURE)` is a good default for `CFG_INSECURE` -- again, assuming it is set to an empty value or "n". If it is not set then "y" should be the default.
Having a warning about deprecation is probably a good idea too. So let's recap:
```
ifneq (undefined,$(flavor CFG_WARN_INSECURE))
$(warn CFG_WARN_INSECURE is deprecated, use CFG_INSECURE instead)
CFG_INSECURE ?= $(CFG_WARN_INSECURE)
else
CFG_INSECURE ?= y
endif
```

Reply:
The issue I see with `CFG_INSECURE ?= $(CFG_WARN_INSECURE)` is the case where one mistakenly sets `CFG_INSECURE` and `CFG_WARN_INSECURE` to different values. When so, I think we should simply trigger a build error.

Triggering an build error when `CFG_WARN_INSECURE` is set, whatever the value, is a another (maybe rude) way to ensure consistency of these insecure OP-TEE core config directives.
`

Reply:
@jenswi-linaro, @jforissier, new proposal:
```c
ifneq (undefined,$(flavor CFG_WARN_INSECURE))
$(info CFG_WARN_INSECURE is deprecated, use CFG_INSECURE instead)
CFG_INSECURE ?= $(CFG_WARN_INSECURE)
# If one of them is enabled, both must match
ifeq (y,$(filter y,$(CFG_INSECURE) $(CFG_WARN_INSECURE)))
ifneq ($(CFG_INSECURE),$(CFG_WARN_INSECURE)
$(error Inconsistent CFG_INSECURE=$(CFG_INSECURE) and CFG_WARN_INSECURE=$(CFG_WARN_INSECURE))
endif
endif
endif # CFG_WARN_INSECURE defined
CFG_INSECURE ?= y

Reply:
> @jenswi-linaro, @jforissier, new proposal:

LGTM

Reply:
It looks a bit complicated, how about:
```
ifneq (undefined,$(flavor CFG_WARN_INSECURE))
$(info CFG_WARN_INSECURE is deprecated, use CFG_INSECURE instead)
CFG_INSECURE ?= $(CFG_WARN_INSECURE)
ifneq ($(CFG_INSECURE),$(CFG_WARN_INSECURE)
$(error Inconsistent CFG_INSECURE=$(CFG_INSECURE) and CFG_WARN_INSECURE=$(CFG_WARN_INSECURE))
endif
endif # CFG_WARN_INSECURE defined
CFG_INSECURE ?= y
```